### ISSUE TITLE
the first character input in the commit message buffer doesn't invoke the input method

### ISSUE THREAD

**Issue by ilupin** [magit/magit#4275]
The problem occurs with Emacs 28.0.50 (2020-12-20). It doesn't appear in Emacs 27.1.

magit version: Magit 20201221.1758, Git 2.25.1, Emacs 28.0.50, gnu/linux

To reproduce the issue:
1. `emacs -Q --debug-init --eval '(setq debug-on-error t)' -L ../dash-20201218.1237/ -L ../git-commit-20201115.2313/ -L ../transient-20201220.1911/ -L ../with-editor-20201030.1232/ -L . -l magit`
2. turn on an input method (e.g. TeX) for text-mode by
  ```
  (setq default-input-method "TeX")
  (add-hook 'text-mode-hook 'toggle-input-method)
  ```
3. in any magit status buffer, press `c c` to create a commit
4. type `\alpha` in the commit message buffer. The first `\` should invoke the input method, but it didn't.

Everything is normal from the second input, e.g., press <kbd>C-g</kbd> first and then type `\alpha` in the commit message buffer. Setup like `(defun git-commit-setup () (activate-input-method "TeX"))` produced the same result.

**Comment by tarsius** [comment]
I cannot reproduce that and I have a hard time imagining how magit/git-commit.el might cause this.

**Comment by ilupin** [comment]
Everything is fine when `COMMIT_EDITMSG` is opened directly, and I think git-commit.el is not causing the problem.

Actually I'm not sure if it is related to magit. I tried to understand the magit/with-editor code executed before git-commit-mode, but it was difficult. I'll try to narrow down the problem anyway.

**Comment by kyleam** [comment]
This sounds like the same as Emacs's [bug#45292][1], which Stefan recently fixed today in ccb3efffc5.  I'm optimistically closing this, but please reopen if I'm mistaken.

[1]: https://debbugs.gnu.org/45292
